agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiComponentInstaller.java

```diff
@@ -128,9 +128,19 @@ private void start(Instrumentation instrumentation) {
     }
 
     File javaTmpDir = new File(System.getProperty("java.io.tmpdir"));
+    boolean readOnlyFileSystem = false;
+    if (javaTmpDir.canRead() && !javaTmpDir.canWrite()) {
+      readOnlyFileSystem = true;
+    }
+
     File tmpDir = new File(javaTmpDir, "applicationinsights-java");
-    if (!tmpDir.exists() && !tmpDir.mkdirs()) {
-      throw new IllegalStateException("Could not create directory: " + tmpDir.getAbsolutePath());
+    if (!readOnlyFileSystem) {
+      if (!tmpDir.exists() && !tmpDir.mkdirs()) {
```

Comment:
Can this be moved down here (to scope it better)?
```suggestion
    if (!readOnlyFileSystem) {
      File tmpDir = new File(javaTmpDir, "applicationinsights-java");
      if (!tmpDir.exists() && !tmpDir.mkdirs()) {
```

Reply:
tmpDir is also used in the else block.